tools/rostopic/src/rostopic/__init__.py

```diff
@@ -1417,6 +1423,8 @@ def _rostopic_cmd_pub(argv):
                       help="read args from YAML file (Bagy)")
     parser.add_option("-l", '--latch', dest="latch", default=False, action="store_true",
                       help="enable latching for -f, -r and piped input.  This latches the first message.")
+    parser.add_option("-s", '--substitute_keywords', dest="substitute_keywords", default=False, action="store_true",
```

Comment:
Please use `--substitute-keywords`. Underscore separators are rather uncommon for command line arguments.